### PR TITLE
Two path list remote test

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses # (issue)

The `remote_files_list` test currently presumes that, if present, all `tag` and `inst_id` combinations are supported by the `list_remote_files` instrument module function. This isn't always the case. Thus, the test was modified to also check for a warning that it is not supported.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide
instructions so we can reproduce. Please also list any relevant details for
your test configuration

- Merged with `load_meta_proc` and tested against the pysatModels `tiegcm` branch. version set locally to 3.0.2 so all tests run and pass.

**Test Configuration**:
* Operating system: MacOS
* Version number: Python 3.9
* Any details about your local setup that are relevant: Merged with `load_meta_proc`

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -112,6 +112,8 @@ This project adheres to [Semantic Versioning](https://semver.org/).
    * Moved function call for Instrument modules init function to end of
      pysat.Instrument instantiation to ensure Instrument is complete when passed
      to init.
+   * Expanded `remote_file_list` to account for warnings raised by
```
Comment (rstoneback) [inline CHANGELOG.md:115]:
```suggestion
   * Expanded `remote_file_list` unit test to account for warnings raised by 
```

File: pysat/tests/classes/cls_instrument_library.py [hunk 2]
```diff
@@ -375,7 +376,7 @@ def test_load(self, clean_level, inst_dict):
         return
 
     @pytest.mark.download
-    def test_remote_file_list(self, inst_dict):
+    def test_remote_file_list(self, inst_dict, caplog):
```
Comment (jklenzing) [inline pysat/tests/classes/cls_instrument_library.py:379]: not sure we need to invoke caplog here if we're not accessing it

File: pysat/tests/classes/cls_instrument_library.py [hunk 3]
```diff
@@ -395,9 +396,19 @@ def test_remote_file_list(self, inst_dict):
...
-            assert len(files) > 0
+
+            # Raise and catch warnings
+            with warnings.catch_warnings(record=True) as war:
+                files = test_inst.remote_file_list(start=date, stop=date,
+                                                   **dl_dict)
+            if len(files) == 0:
+                # Not all `tag` or `inst_id` settings may have support.
+                # If not supported, warning should be issued.
+                testing.eval_warnings(war, 'Not implemented in this version.',
```
Comment (rstoneback) [inline pysat/tests/classes/cls_instrument_library.py:407]:
```suggestion
                testing.eval_warnings(war, 'Not implemented',
```

**Review by jklenzing — COMMENTED** [review]
The code should take care of this already.  The issue is being caused by wrong settings for the test flags in the tiegcm pull.  See comments there for suggestions.

This does stem from a bug in the existing logic, and should be updated either in the default logic or the `generate_instrument_list` function in case users are overenthusiastic in changing default flag values.

**Comment by jklenzing** [comment]
Here's the relevant bit in the code:
https://github.com/pysat/pysat/blob/6389d9d7fef5e7ca234eae5e18ee8285326476f2/pysat/utils/_core.py#L465-L486

I've verified that the two tiegcm instruments are being sorted correctly, not sure why the test is failing.

**Comment by jklenzing** [comment]
Just realized this isn't a bugfix, but expansion of tests.  The key is to create a test for the warning message.  I'd suggest modifying the download warning test instead of the proposed change here:
https://github.com/pysat/pysat/blob/6389d9d7fef5e7ca234eae5e18ee8285326476f2/pysat/tests/classes/cls_instrument_library.py#L406-L427

If there is no download support, it needs the pytest mark for no_download to be run.

**Pull request closed without merging**